cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/partitioned/PartitionedFileSetDataset.java

```diff
@@ -839,6 +848,8 @@ public String getOutputFormatClassName() {
       outputArgs.put(Constants.Dataset.Partitioned.HCONF_ATTR_OUTPUT_FORMAT_CLASS_NAME,
                      files.getOutputFormatClassName());
       outputArgs.put(Constants.Dataset.Partitioned.HCONF_ATTR_OUTPUT_DATASET, getName());
+    } else {
+      assertNotExists(outputKey);
```

Comment:
I think this may break existing code. Previously, it was not requiring a transaction, now it does. 
- if existing code calls this outside a transaction, it will break
- the method now needs to be annotated with @ReadOnly because it accesses data